File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int left=0,right=0,len=0;
        unordered_map<char,int> mp;
        for(int i=0;i<s.size();i++){
            if(mp.find(s[right])!=mp.end()){
                left=max(mp[s[right]]+1,left);
            }
            mp[s[right]]=right;
            len=max((right-left)+1,len);
            right++;
        }
        return len;
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        HashMap<Character,Integer> mpp = new HashMap<>();
        int left=0,right=0;
        int len=0;
        int n=s.length();
        while(right<n){
            if(mpp.containsKey(s.charAt(right))){
                left=Math.max(mpp.get(s.charAt(right)) + 1,left);
            }
            len=Math.max((right-left+1),len);
            mpp.put(s.charAt(right),right);
            right++;
        }
        return len;
        
    }
}